### PR TITLE
(SC-10) Readme

### PR DESCRIPTION
**Pull request by RasussRogue**
**Motivation**
 
- To add a Readme to the project
 
**Modification**
 
- Added a Readme explaining the goal, choices, technical tools and commands required to use the project.
 
**Remarks**
 
- Will be updated regularly as the project moves forward.

### CODE REVIEW

**Comment by tkuy** [comment]
Pls squash your commits. 6 commits with the same names should be the same

File: README.md [hunk 1]
```diff
@@ -1 +1,56 @@
-# ReactChamps
+# Stat Check
+
+[Lien du Trello](https://trello.com/b/uc8WwvZB/stat-check)
+
+## Goal
+This project is a way for me to practice and perfect my React programming skills. It is a learning tool that allows me to experiment and try various things out when it comes to front-end development. It also allows me to discover new technologies and learn to master them, all based around a theme that personnaly interests me : video games, and particularly League of Legends.
```
Comment (tkuy) [inline README.md:6]: It also has a purpose to learn to work on agile method

File: README.md [hunk 2]
```diff
@@ -1 +1,56 @@
...
+| MDI (Material Design Icons |5.9.55|https://materialdesignicons.com/|
+| React |17.0.1|https://fr.reactjs.org/|
+| React Router |5.2.0|https://reactrouter.com/web/guides/quick-start|
+| Typescript |4.1.5|https://www.typescriptlang.org/docs/|
+| Webpack |5.21.2| https://webpack.js.org/concepts/ |
+| Webpack Dev Server|3.11.2|https://webpack.js.org/configuration/dev-server/|
+
+## Choices
+
+I went for the use of **React Hooks** instead of class based components due to their general ease of execution concerning states (useState and useEffect in particular making it much simpler to program). Furthermore, being in a phase of learning React, I would rather start with a more efficient and powerful method of development without getting used to class-based component first, as it will almost force me to un-learn certain habits once I would switch to hooks.
```
Comment (tkuy) [inline README.md:23]: good. Well said

File: README.md [hunk 3]
```diff
@@ -1 +1,56 @@
...
+
+I choose to use **Material-UI** for its ease of setup and utilisation, which allows me to focus more on learning React and other techs rather than spend too much time on style etc...
```
Comment (tkuy) [inline README.md:25]: Actually, it is important for you to learn with libraries like material. And it also gives you what you can't learn through just react, and it is about choices of styles, visual concepts, colors and user experience. Keep in mind that you will not be just a "developer" but also, someone who will be able to choose colors, design, make mocks and be able to choose the best components to improve your user experience.

File: README.md [hunk 4]
```diff
@@ -1 +1,56 @@
...
+
+I also decided to go for **Typescript** over native Javascript, as I have the background of a OOP developer (I mostly studied programming through Java) and thus feel more familiar and performant with TypeScript.
```
Comment (tkuy) [inline README.md:27]: I dont like this paragraph

File: README.md [hunk 5]
```diff
@@ -1 +1,56 @@
...
+
+I decided to use **Webpack** for an easier loading of my assets (such as CSS and TS), as well as a clearer config method (through the webpack.config.js file).
```
Comment (tkuy) [inline README.md:29]: Little bit poor

File: README.md [hunk 6]
```diff
@@ -1 +1,56 @@
...
+
+I also decided solely use **external APIs** (Riot Games DataDragon for example) as my goal for this project is not to set up my own backend (which I could), but rather to focus on learning front-end development.
+
+For the color choice, I based myself off of [**this article**](https://www.smashingmagazine.com/2016/04/web-developer-guide-color/), which I found both helpful and simple to apply. I ended up with the following colors for my app :
+
+| Color      | Hexa        |
+| ---        | ---         |
+| Primary    | **#17172E** |
```
Comment (tkuy) [inline README.md:37]: Is it updated ?

File: README.md (hunk 1 above)
Comment (RasussRogue) [reply README.md:6]: Yes, I forgot to mention the organization and mostly time-management aspect of the learning process.

File: README.md (hunk 6 above)
Comment (RasussRogue) [reply README.md:37]: Updated them just before the PR, yes.

**Pull request merged**